frappe/utils/connections.py

```diff
@@ -2,15 +2,21 @@
 from urllib.parse import urlparse
 
 from frappe import get_conf
+from frappe.exceptions import UrlSchemeNotSupported
 
 REDIS_KEYS = ("redis_cache", "redis_queue")
 
 
-def is_open(ip, port, timeout=10):
-	s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+def is_open(scheme, hostname, port, timeout=10):
+	if scheme in ["redis", "rediss", "postgres", "mariadb"]:
```

Comment:
```suggestion
	if scheme in ["redis", "postgres", "mariadb"]:
```

Reply:
Technically, it's [provsional](https://www.iana.org/assignments/uri-schemes/prov/rediss), yes.

Theoretically, the python and java-script libraries would support it.

Practically, I haven't tested if it would _actually_ work.